promtimer/cbstats.py

```diff
@@ -429,14 +435,36 @@ def get_stats_sources(cluster, user, password):
                 if host is None:
                     host = '127.0.0.1'
                 services = node['services']
-                port = services['mgmt']
-                source = ServerNode(host, port, user, password)
+                port = services['mgmtSSL'] if secure else services['mgmt']
+                source = ServerNode(host, port, user, password, secure)
                 result.append(source)
             return result
         except OSError as err:
             logging.error('error: can\'t access cluster: {}'.format(err))
             return []
 
+    @staticmethod
+    def get_stats_sources_from_nodes(nodes, user, password):
+        result = []
+        for node in nodes:
+            secure = node.startswith("https://")
+            response = util.execute_request(node, 'nodes/self',
+                                            username=user, password=password)
+            nodes_self = json.loads(response.read())
+            real_hostname = nodes_self.get('hostname')
+            if real_hostname is None:
+                real_hostname = '127.0.0.1'
+            else:
+                real_hostname = real_hostname.rstrip(':8091')
+            real_port = nodes_self['ports']['httpsMgmt'] if secure else nodes_self['ports']['mgmt']
+            short_name = f'{real_hostname}:{real_port}'
+            host_and_port = node.split('://')[1]
+            host, port = host_and_port.split(':')
```

Comment:
This requires a host and port. Would be nice to default to 8091 if http and 18091 if the scheme is TLS.

Reply:
Default to 8091 and 18091 now